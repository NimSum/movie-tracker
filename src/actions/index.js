export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
})

export const formType = (type) => ({
  type: 'FORM_TYPE',
  form: type
})

export const currentUser = (id, name, email, favorites) => ({
  type: 'CURRENT_USER',
  userInfo: { id, name, email, favorites }
})

export const selectedMovie = (movie) => ({
  type: 'SELECTED_MOVIE',
  movie
})

export const updateFavorites = (favorite) => ({
  type: 'UPDATE_FAVORITES',
  favorite
})