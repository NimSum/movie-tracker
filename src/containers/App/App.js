import React from 'react';
import './App.scss';
import Header from '../Header'
import { Route, NavLink, Link, Switch, Redirect } from 'react-router-dom';

import movie from './movie.jpg'

function App() {
  let all = Array(1000).fill('')
  const movieTiles = all.map(space => {
    return <div className='tile'>
      <img src={movie} className='tile'/>
    </div>
  })
  return (
    <div className="App">
      <Header />
      {movieTiles}
    </div>
  );
}

export default App;
