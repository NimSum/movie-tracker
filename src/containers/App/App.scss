.App {
  text-align: center;
  display: grid;
  grid-template-columns: repeat(30, 1fr);
}

.tile {
  position: relative;
  filter: brightness(60%);
  // max-height: 2px;
  object-fit: cover;
  width: 100%;
  z-index: 0;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}