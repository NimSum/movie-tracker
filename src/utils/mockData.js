export const mockLatestMovie = [{
  "adult": false,
  "backdrop_path": null,
  "belongs_to_collection": null,
  "budget": 0,
  "genres": [],
  "homepage": null,
  "id": 600900,
  "imdb_id": null,
  "original_language": "fr",
  "original_title": "Le Cri nu",
  "overview": "",
  "popularity": 0,
  "poster_path": null,
  "production_companies": [],
  "production_countries": [{
    "iso_3166_1": "FR",
    "name": "France"
  }],
  "release_date": "1979-01-01",
  "revenue": 0,
  "runtime": null,
  "spoken_languages": [],
  "status": "Released",
  "tagline": "",
  "title": "Le Cri nu",
  "video": false,
  "vote_average": 0,
  "vote_count": 0
}]

export const mockUser = {
  id: 1,
  name: "misskelly",
  email: "kelly@kelly.com",
  favorites: [{
    id: 2,
    movie_id: 287947,
    user_id: 1,
    title: "Shazam!",
  }]
}

export const mockMovies = [
  {
    movie_id: 299534,
    title: "Avengers: Endgame",
    poster_path: "/or06FN3Dka5tukK1e9sl16pB3iy.jpg",
    release_date: "2019-04-24",
    vote_average: 8.6,
    overview: "After the devastating events ...."
  },
  {
    movie_id: 447404,
    title: "Pokémon Detective Pikachu",
    poster_path: "/wgQ7APnFpf1TuviKHXeEe3KnsTV.jpg",
    release_date: "2019-05-03",
    vote_average: 7.1,
    overview: "In a world where people collect pocket-size monsters (Pokémon) to do battle...."
  }
]

export const mockSelectedMovie = {
  adult: false,
  backdrop_path: "/bi4jh0Kt0uuZGsGJoUUfqmbrjQg.jpg",
  belongs_to_collection: null,
  budget: 80000000,
  homepage: "http://www.shazammovie.com",
  id: 287947,
  imdb_id: "tt0448115",
  original_language: "en",
  original_title: "Shazam!",
  overview: "A boy is given the ability to become an adult superhero in times of need with a single magic word.",
  popularity: 307.286,
  poster_path: "/xnopI5Xtky18MPhK40cZAGAOVeV.jpg",
  release_date: "2019-03-23",
  revenue: 321937864,
  runtime: 132,
  status: "Released",
  tagline: "Just Say The Magic Word",
  title: "Shazam!",
  video: false,
  vote_average: 7.2,
  vote_count: 1528,
  showFavorites: false
}