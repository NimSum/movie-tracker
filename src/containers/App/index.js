import React, { Component } from 'react';
import './App.scss';
// import Header from '../Header'
import { Route, NavLink, Link, Switch, Redirect } from 'react-router-dom';
import { mockLatestMovie } from '../../utils/mockData'
import movie from './movie.jpg'
import { addMovies } from '../../actions/index'
import { connect } from 'react-redux'
import { key } from '../../apiKey';
import { fetchMovies } from '../../utils/apiFetches/fetchMovies';

export class App extends Component {
  constructor() {
    super();
  }

  async componentDidMount() {
    const nowShowingUrl = `https://api.themoviedb.org/3/movie/now_playing?api_key=${key}&language=en-US&page=1`;
    try {
      const movies = await fetchMovies(nowShowingUrl);
      await this.props.addMovies([movies])
    } catch(error) {
      console.log(error)
    }
  }

  render() {
    let all = Array(1000).fill('')
    const movieTiles = all.map(space => {
      return <div className='tile'>
        <img src={movie} className='tile'/>
      </div>
    })
    
    return (
      <div className="App">
        {/* <Header /> */}
        {movieTiles}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
