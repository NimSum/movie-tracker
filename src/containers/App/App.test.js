import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from '../App';
import { shallow } from 'enzyme'
import { mockMovies } from '../../utils/mockData'
import { addMovies } from '../../actions/index'


jest.mock('../../actions/index')

describe('App', () => {
  let wrapper
  let mockAddMovies
  beforeEach(() => {

    mockAddMovies = jest.mock()
    wrapper = shallow(
      <App 
        movies={mockMovies}
        addMovies={mockAddMovies}
       />
    )
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      ok: true,
      status: 200,
      json: () => Promise.resolve(mockMovies)
    }))
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});

describe('mapStateToProps', () => {
  it('should return a props object with movies', () => {
    const mockState = { movies: mockMovies}
    const expected = { movies: mockState.movies }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)

  });
});

describe('mapDispatchToProps', () => {
  
  it('calls dispatch with add movies action', () => {
    const dispatch = jest.fn()
    const actionToDispatch = addMovies(mockMovies)  
    mapDispatchToProps(dispatch(actionToDispatch))
    expect(dispatch).toHaveBeenCalledWith(actionToDispatch);
  });
    
});
