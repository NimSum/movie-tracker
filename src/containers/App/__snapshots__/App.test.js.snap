// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <ConnectFunction />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/favorites"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(LoginPage)",
        "type": [Function],
      }
    }
    exact={true}
    path="/login"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(MoviePage)",
        "type": [Function],
      }
    }
    exact={true}
    path="/movies/:id"
  />
</div>
`;
